### PR TITLE
vmgenid: check that vmgenid (in qemu.conf) is different than `volatile.uuid.generation`

### PR DESCRIPTION
**Pull request by gabrielmougard**
* Test associated with https://github.com/lxc/lxd/pull/11498

TEST : 

* side loading (see here https://paste.ubuntu.com/p/6H7fW2kM9p/) and manual testing OK

### CODE REVIEW

File: bin/test-lxd-storage-vm [hunk 1]
```diff
@@ -431,7 +431,8 @@
         # Check that the volatile.uuid.generation setting is applied to the QEMU process.
         vmGenID=$(lxc config get v1 volatile.uuid.generation)
         vmPID=$(lxc info v1 | awk '/^PID:/ {print $2}')
-        qemuGenID=$(sed -n 's/.*vmgenid,guid=\([0-9a-fA-F]\{8\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{12\}\).*/\1/p' "/proc/${vmPID}/cmdline")
+        qemuConfFile=$(sed -n 's/.*-readconfig\(.*qemu.conf\).*/\1/p' "/proc/${vmPID}/cmdline" | tr '\0' '\n')
+        qemuGenID=$(awk '/driver = "vmgenid"/,/guid = / {print $3}' $qemuConfFile | sed -n 's/"\([0-9a-fA-F]\{8\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{12\}\)"/\1/p')
```
[the remark this answers is not in the document]
Comment (tomponline) [reply bin/test-lxd-storage-vm:435]: CC @gabrielmougard

**Comment by gabrielmougard** [comment]
@tomponline it's probably simple but when I do : 

`qemuGenID=$(awk '/driver = "vmgenid"/,/guid = / {print $3}' "${qemuConfFile}" | sed -n 's/"\([0-9a-fA-F]\{8\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{12\}\)"/\1/p')`

I end up having : `awk: cannot open 
/var/snap/lxd/common/lxd/logs/v1/qemu.conf (No such file or directory)` whereas I'm pretty the fil exists. And without the double quotes it works just fine...

**Comment by gabrielmougard** [comment]
For now, I added a `shellcheck disable=SC2086` (Double quote to prevent globbing and word splitting.)

**Comment by tomponline** [comment]
> For now, I added a `shellcheck disable=SC2086` (Double quote to prevent globbing and word splitting.)

"for now"?

**Comment by tomponline** [comment]
I think we should just hardcode `/var/snap/lxd/common/lxd/logs/v1/qemu.conf` as it will always be there for this test.

**Comment by tomponline** [comment]
@gabrielmougard 

> ./bin/test-lxd-storage-vm:433:9: warning[SC2034]: vmPID appears unused. Verify use (or export if used externally).

**Review by tomponline — APPROVED** [review]

**Pull request merged**